modules/migration/file_format.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+	"gopkg.in/yaml.v2"
+)
+
+// Load project data from file, with optional validation
+func Load(filename string, data interface{}, validation bool) error {
+	isJSON := strings.HasSuffix(filename, ".json")
+
+	_, err := os.Stat(filename)
```

Comment:
Why is stat necessary? If this is purely to check if the file exists - `ReadFile` should give a error for that as well.

Reply:
You are correct, removed.